src/irr/asset/CImageLoaderTGA.cpp

```diff
@@ -301,59 +301,135 @@ asset::SAssetBundle CImageLoaderTGA::loadAsset(io::IReadFile* _file, const asset
 
 #else
 
-	switch (header.ImageType)
+	const uint64_t imageSize = header.ImageHeight * header.ImageWidth * header.PixelDepth / 8;
+	auto imageBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(imageSize);
+
+	uint8_t* data = static_cast<uint8_t*>(imageBuffer->getPointer());
+
+	ICPUImage::SCreationParams imgInfo;
+	imgInfo.type = ICPUImage::ET_2D;
+	imgInfo.extent.width = header.ImageWidth;
+	imgInfo.extent.height = header.ImageHeight;
+	imgInfo.extent.depth = 1u;
+	imgInfo.mipLevels = 1u;
+	imgInfo.arrayLayers = 1u;
+	imgInfo.samples = ICPUImage::ESCF_1_BIT;
+	imgInfo.flags = static_cast<IImage::E_CREATE_FLAGS>(0u);
+
+	switch (header.PixelDepth)
 	{
-	case 1: // uncompressed color-mapped image
-	case 2: // uncompressed rgb image
-	case 3: // uncompressed grayscale image
+	case 8:
+	{
+		if (header.ImageType != 3)
+		{
+			os::Printer::log("Loading 8-bit non-grayscale is NOT supported.", ELL_ERROR);
+			return {};
+		}
 
-	//it is temporary and for testing purpose
-	case 10:
+		imgInfo.format = asset::EF_R8_SRGB;
```

Comment:
this format is not supported widely, this is why I had a function here originally that would convert R8_SRGB to RGB8_SRGB in software.

Reply:
@devshgraphicsprogramming in that should I calculate average of channels and perform grayscale or should I fill `g` and `b` channels with certain default values like 0?

Reply:
I see we have someting like `video::convertColor`, but I'm not sure if it'll solve it though

Reply:
no what  Imean is that you should duplicare the R8_SRGB channel to fill RGB8_SRGB

so its basically is:
```glsl
out.rgb = in.rrr;
```

Reply:
okay, great